extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala

```diff
@@ -1687,6 +1687,26 @@ class HiveCatalogPrivilegeBuilderSuite extends PrivilegesBuilderSuite {
     assert(pi4.columns.size === 2)
     assert(pi4.columns === Seq("key", "pid"))
   }
+
+  test("KYUUBI #4533: Cols extracted by authZ should resolve caseSensitive") {
+    val plan1 = sql(s"SELECT key, pid FROM $reusedPartTable WHERE PID = '1'")
+      .queryExecution.optimizedPlan
+    val (in1, out1, _) = PrivilegesBuilder.build(plan1, spark)
+    assert(in1.size === 1)
+    assert(out1.isEmpty)
+    val pi1 = in1.head
+    assert(pi1.columns.size === 2)
+    assert(pi1.columns === Seq("key", "PID"))
```

Comment:
What is the exact rules of extracting the column name, either conform to runtime declared in SQL or to the original name in source table.

Based on my plain understanding to the word `caseSensitive`, it that ["key", "pid", "PID"] should be the the correct output columns of the query "SELECT key, pid FROM $reusedPartTable WHERE PID = '1'" ? Here the output with  ["key", "PID"] reflects "pid" is overritten by "PID".

But curiously the original column name in source table of the test is `pid` , all in lower case.

And if you are following the declared case sensitive way in SQL, please also other cases should be discussed and tested, including the mixed combinationof of uppercase and lowercase letters.

Reply:
As for me, I would prefer to use the original name of the source table instead of case-sensitive declared column in runtime SQL.

Reply:
Yes, we should keep consistent with raw table.

Reply:
SGTM.